.github/workflows/tests.yml

```diff
@@ -175,7 +175,7 @@ jobs:
       if: matrix.os == 'ubuntu-latest'
       run: |
         brew install hadolint
-        hadolint Dockerfile
+        hadolint Dockerfile docker/*/Dockerfile
 
     - name: Build Docker image
```

Comment:
What's this doing compared to the stuff below?

Reply:
Same same but different. Building the image is pretty much the same (with a base image of Ubuntu 16.04 LTS). These two jobs could be merged into one.

Reply:
Yeh, that'd be ideal I think to consolidate as much as possible.

Reply:
I can look into that. One difference for Ubuntu 16.04 is that it needs a newer version of Git installed from a PPA. Should be able to make that conditional.

Reply:
Turns out not to be a problem. The version of Git in Ubuntu 16.04 is sufficient. There is now only one Dockerfile for all three versions of Ubuntu.